src/dotnet/MsbuildProject.cs

```diff
@@ -166,12 +174,97 @@ public static void EnsureAllReferencesExist(List<string> references)
             }
         }
 
+        public IEnumerable<NuGetFramework> GetTargetFrameworks()
+        {
+            if (_cachedTfms != null)
+            {
+                return _cachedTfms;
+            }
+
+            var project = GetEvaluatedProject();
+
+            var properties = project.AllEvaluatedProperties
+                                    .Where(p => p.Name.Equals("TargetFrameworks", StringComparison.OrdinalIgnoreCase) ||
+                                                p.Name.Equals("TargetFramework", StringComparison.OrdinalIgnoreCase))
+                                    .Select(p => p.EvaluatedValue.ToLower()).ToList();
+
+            var uniqueTfms = new HashSet<string>();
+
+            foreach (var property in properties)
+            {
+                var tfms = property
+                                .Split(';')
+                                .Select((tfm) => tfm.Trim())
+                                .Where((tfm) => !string.IsNullOrEmpty(tfm));
+
+                foreach (var tfm in tfms)
+                {
+                    uniqueTfms.Add(tfm);
+                }
+            }
+
+            _cachedTfms = uniqueTfms.Select((frameworkString) => NuGetFramework.Parse(frameworkString)).ToList();
+            return _cachedTfms;
+        }
+
+        public bool CanWorkOnFramework(NuGetFramework framework)
```

Comment:
This method and the next should likely go as extensions to NuGetFrameowkr as well, receiving a list of the framework as parameter.